pkg/app/server.go

```diff
@@ -66,10 +70,20 @@ func (pl promLogger) Log(v ...interface{}) error {
 	return nil
 }
 
-// RunKubeStateMetrics will build and run the kube-state-metrics.
+// RunKubeStateMetricsWrapper runs KSM with context cancellation.
+func RunKubeStateMetricsWrapper(ctx context.Context, opts *options.Options, factories ...customresource.RegistryFactory) error {
+	err := runKubeStateMetrics(ctx, opts, factories...)
+	if ctx.Err() == context.Canceled {
+		klog.Infoln("RESTARTING kube-state-metrics")
```

Comment:
Decided to make this uppercase, to make it sort of "pop out" in the sea of logs, since I thought it made sense for the reader to be able to know where the older logs ended and where the newer ones started from.